File: csUdp/Chat.Common/C2S.Proxy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Icet.Message;
using System.Net.Sockets;
using Newtonsoft.Json;
using Chat.Common;
namespace C2S
{
	public class Proxy : MessageProxy
	{
		public const int Version = 100;

		public bool Heartbeat(UdpClient client, String uid)
		{
			if (client == null) return false;
			Message.Heartbeat msg = new Message.Heartbeat();
			msg.id = "100";
			msg.uid = uid;
			string jsonmsg = JsonConvert.SerializeObject(msg);
			byte[] data = UTF8Encoding.UTF8.GetBytes(jsonmsg);
			client.Send(data, data.Length);
			return true;
		}
		public bool ReqChat(UdpClient client, String uid, String group, String chat)
		{
			if (client == null) return false;
			Message.ReqChat msg = new Message.ReqChat();
			msg.id = "101";
			msg.uid = uid;
			msg.group = group;
			msg.chat = chat;
			string jsonmsg = JsonConvert.SerializeObject(msg);
			byte[] data = UTF8Encoding.UTF8.GetBytes(jsonmsg);
			client.Send(data, data.Length);
			return true;
		}
		public bool ReqLogin(UdpClient client, String uid)
		{
			if (client == null) return false;
			Message.ReqLogin msg = new Message.ReqLogin();
			msg.id = "102";
			msg.uid = uid;
			string jsonmsg = JsonConvert.SerializeObject(msg);
			byte[] data = UTF8Encoding.UTF8.GetBytes(jsonmsg);
			client.Send(data, data.Length);
			return true;
		}
		public bool ReqLogout(UdpClient client, String uid)
		{
			if (client == null) return false;
			Message.ReqLogout msg = new Message.ReqLogout();
			msg.id = "103";
			msg.uid = uid;
			string jsonmsg = JsonConvert.SerializeObject(msg);
			byte[] data = UTF8Encoding.UTF8.GetBytes(jsonmsg);
			client.Send(data, data.Length);
			return true;
		}
		public bool ReqJoin(UdpClient client, String uid, String group)
		{
			if (client == null) return false;
			Message.ReqJoin msg = new Message.ReqJoin();
			msg.id = "104";
			msg.uid = uid;
			msg.group = group;
			string jsonmsg = JsonConvert.SerializeObject(msg);
			byte[] data = UTF8Encoding.UTF8.GetBytes(jsonmsg);
			client.Send(data, data.Length);
			return true;
		}
		public bool ReqLeave(UdpClient client, String uid, String group)
		{
			if (client == null) return false;
			Message.ReqLeave msg = new Message.ReqLeave();
			msg.id = "105";
			msg.uid = uid;
			msg.group = group;
			string jsonmsg = JsonConvert.SerializeObject(msg);
			byte[] data = UTF8Encoding.UTF8.GetBytes(jsonmsg);
			client.Send(data, data.Length);
			return true;
		}
		public bool ReqUserList(UdpClient client, String uid, String group)
		{
			if (client == null) return false;
			Message.ReqUserList msg = new Message.ReqUserList();
			msg.id = "106";
			msg.uid = uid;
			msg.group = group;
			string jsonmsg = JsonConvert.SerializeObject(msg);
			byte[] data = UTF8Encoding.UTF8.GetBytes(jsonmsg);
			client.Send(data, data.Length);
			return true;
		}
	}
}

File: csUdp/Chat.Common/c2s.proxy.js
var c2smessage = require('./c2s.message.js');
var entity = require('./entity.js');
var moment = require('moment');

var proxy = function() {
};

proxy.prototype = {
	Heartbeat : function(to_ip, to_port, server, uid) {
		try {
			var send_message = new c2smessage.HeartbeatMessage();
			send_message.id = "100";
			send_message.uid = uid;

			var send_data = new Buffer(JSON.stringify(send_message));
			server.send(send_data, 0, send_data.length, to_port, to_ip);
		} catch(e) {
			console.log(e);
		}
	},
	ReqChat : function(to_ip, to_port, server, uid, group, chat) {
		try {
			var send_message = new c2smessage.ReqChatMessage();
			send_message.id = "101";
			send_message.uid = uid;
			send_message.group = group;
			send_message.chat = chat;

			var send_data = new Buffer(JSON.stringify(send_message));
			server.send(send_data, 0, send_data.length, to_port, to_ip);
		} catch(e) {
			console.log(e);
		}
	},
	ReqLogin : function(to_ip, to_port, server, uid) {
		try {
			var send_message = new c2smessage.ReqLoginMessage();
			send_message.id = "102";
			send_message.uid = uid;

			var send_data = new Buffer(JSON.stringify(send_message));
			server.send(send_data, 0, send_data.length, to_port, to_ip);
		} catch(e) {
			console.log(e);
		}
	},
	ReqLogout : function(to_ip, to_port, server, uid) {
		try {
			var send_message = new c2smessage.ReqLogoutMessage();
			send_message.id = "103";
			send_message.uid = uid;

			var send_data = new Buffer(JSON.stringify(send_message));
			server.send(send_data, 0, send_data.length, to_port, to_ip);
		} catch(e) {
			console.log(e);
		}
	},
	ReqJoin : function(to_ip, to_port, server, uid, group) {
		try {
			var send_message = new c2smessage.ReqJoinMessage();
			send_message.id = "104";
			send_message.uid = uid;
			send_message.group = group;

			var send_data = new Buffer(JSON.stringify(send_message));
			server.send(send_data, 0, send_data.length, to_port, to_ip);
		} catch(e) {
			console.log(e);
		}
	},
	ReqLeave : function(to_ip, to_port, server, uid, group) {
		try {
			var send_message = new c2smessage.ReqLeaveMessage();
			send_message.id = "105";
			send_message.uid = uid;
			send_message.group = group;

			var send_data = new Buffer(JSON.stringify(send_message));
			server.send(send_data, 0, send_data.length, to_port, to_ip);
		} catch(e) {
			console.log(e);
		}
	},
	ReqUserList : function(to_ip, to_port, server, uid, group) {
		try {
			var send_message = new c2smessage.ReqUserListMessage();
			send_message.id = "106";
			send_message.uid = uid;
			send_message.group = group;

			var send_data = new Buffer(JSON.stringify(send_message));
			server.send(send_data, 0, send_data.length, to_port, to_ip);
		} catch(e) {
			console.log(e);
		}
	},
};

exports.proxy = proxy;
